casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -826,12 +830,27 @@ object BlockAPI {
                            .pure[F]
                        case None =>
                          for {
-                           lastFinalizedBlock <- casper.lastFinalizedBlock
-                           runtimeManager     <- casper.getRuntimeManager
-                           postStateHash      = ProtoUtil.postStateHash(lastFinalizedBlock)
-                           res                <- runtimeManager.playExploratoryDeploy(term, postStateHash)
-                           lightBlockInfo     <- getLightBlockInfo[F](lastFinalizedBlock)
-                         } yield (res, lightBlockInfo).asRight[Error]
+                           targetBlock <- if (blockHash.isEmpty)
+                                           casper.lastFinalizedBlock.map(_.some)
+                                         else
+                                           BlockStore[F].get(
+                                             ByteString.copyFrom(Base16.unsafeDecode(blockHash.get))
```

Comment:
This will thrown error if hash string is not valid. I know there is no elegant solution but please check if just `raiseError` with some meaningful message is enough to be reported correctly to the client.